### DISCUSSION TITLE
Create a custom node view that cannot be deleted

### DISCUSSION THREAD

**Discussion by tducloc** [Milkdown/milkdown#1334] [Q&A] [answered]
Hi, is it possible to create a nodeViewFactory component that cannot be deleted? I've searched the Milkdown website, including the code and documentation, but couldn't find any information on this.

This is one of my sample code

`$view(mathInlineSchema.node, () =>
        nodeViewFactory({
          component: Math,
          stopEvent: () => true,
        })
      )`

**Comment by Saul-Mirone** [comment, marked as answer]
It's possible but it's not really about how to build the node view. What you really need is [filterTransaction](https://prosemirror.net/docs/ref/#state.PluginSpec.filterTransaction). You can take a look at the [discussion here](https://discuss.prosemirror.net/t/how-to-prevent-node-deletion/130/8).

**Reply by tducloc** [reply]
Thank you for your solution! I was wondering if you could kindly let me know where I should use filterTransaction to prevent the node from being deleted?

**Reply by Saul-Mirone** [reply]
```ts
import { $prose } from '@milkdown/utils'
import { Plugin } from '@milkdown/prose/state'
const myPlugin = $prose((ctx) => {
  return new Plugin({
    filterTransaction: (tr) => {
    }
  })
})
```

**Reply by tducloc** [reply]
Thank you very much.
